### PR TITLE
Add Microsoft OMS linux agent helm chart

### PR DESCRIPTION
**Pull request by samisms**

### CODE REVIEW

**Review by unguiculus — CHANGES_REQUESTED** [review]
Thanks for the PR. Please check our best practices, especially regarding naming resources and labels.

https://github.com/kubernetes/helm/blob/master/docs/chart_best_practices/labels.md

Also, see chart examples for reference: https://github.com/kubernetes/helm/tree/master/docs/examples

BTW, `helm create` with Helm 2.5+ creates an initial experience with best practices applied.

**Comment by samisms** [comment]
@unguiculus Modified chart and template to follow best practices . Squashed commits. Please review and let me know if any further changes are required. Thanks.

File: stable/msoms/Chart.yaml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+apiVersion: v1
+description: A chart for deploying omsagent as a daemonset Kubernetes
+name: msoms
+version: 0.1.0
```
Comment (unguiculus) [inline stable/msoms/Chart.yaml:4]: Add `appVersion`. Is there an `icon` you could add?

File: stable/msoms/Chart.yaml [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+home: https://azure.microsoft.com/en-us/services/log-analytics/
+sources:
+  - https://github.com/Microsoft/OMS-docker
+  - https://github.com/Microsoft/OMS-Agent-for-Linux
+  - https://github.com/Microsoft/Docker-Provider
+maintainers:
+  - name: OMS Containers
```
Comment (unguiculus) [inline stable/msoms/Chart.yaml:11]: Can you add a github username?

File: stable/msoms/README.md [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+
+## Introduction
+
+This chart deploys an OMS sidecar agent as a daemonset on a [Kubernetes](http://kubernetes.io) cluster using the [Helm](https://helm.sh) package manager. The OMS agent enables rich and real-time analytics for Docker containers. With this solution, you can see which containers are running on your container hosts and what images are running in the containers. You can view detailed audit information showing commands used with containers. And, you can troubleshoot containers by viewing and searching centralized logs without having to remotely view Docker or hosts. You can find containers that may be noisy and consuming excess resources on a host. And, you can view centralized CPU, memory, storage, and network usage and performance information for containers. For more information refer to the [documentation](https://docs.microsoft.com/en-us/azure/log-analytics/log-analytics-containers).
+
+## Prerequisites
+
+- Kubernetes 1.4+ with Beta APIs enabled
```
Comment (unguiculus) [inline stable/msoms/README.md:11]: Let's make it 1.6+

File: stable/msoms/templates/omsagent-daemonset.yaml [hunk 4]
```diff
@@ -0,0 +1,55 @@
+apiVersion: extensions/v1beta1
+kind: DaemonSet
+metadata:
+ name: {{ .Values.agentname }}
```
Comment (unguiculus) [inline stable/msoms/templates/omsagent-daemonset.yaml:4]: Should be `{{ template "fullname" . }} `

File: stable/msoms/templates/omsagent-daemonset.yaml [hunk 5]
```diff
@@ -0,0 +1,55 @@
+apiVersion: extensions/v1beta1
+kind: DaemonSet
+metadata:
```
Comment (unguiculus) [inline stable/msoms/templates/omsagent-daemonset.yaml:3]: Labels are missing.

File: stable/msoms/templates/omsagent-daemonset.yaml [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+ name: {{ .Values.agentname }}
+spec:
+ updateStrategy:
+    type: RollingUpdate
+ template:
+  metadata:
+   labels:
```
Comment (unguiculus) [inline stable/msoms/templates/omsagent-daemonset.yaml:10]: Only `app` and `release` label should go here.

File: stable/msoms/values.yaml [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+# This is a YAML-formatted file.
+# Declare variables to be passed into your templates.
+agentname: omsagent
+secretname: omsagent-secret
+omsagent:
+  image:
+    registry:
+    org: microsoft
+    tag: latest
+    pull_policy: Always
```
Comment (unguiculus) [inline stable/msoms/values.yaml:11]: Camel case, please. No underscores.

File: stable/msoms/values.yaml [hunk 8]
```diff
@@ -0,0 +1,23 @@
+# Default values for omsagent.
+# This is a YAML-formatted file.
+# Declare variables to be passed into your templates.
```
Comment (unguiculus) [inline stable/msoms/values.yaml:3]: The whole file is incorrectly indented.

File: stable/msoms/values.yaml [hunk 9]
```diff
@@ -0,0 +1,23 @@
...
+agentname: omsagent
+secretname: omsagent-secret
+omsagent:
+  image:
+    registry:
```
Comment (unguiculus) [inline stable/msoms/values.yaml:8]: Please drop `registry` and use a more common way of specifying image.

```yaml
image:
  repository:
  tag:
  pullPolicy:
```

File: stable/msoms/values.yaml [hunk 10]
```diff
@@ -0,0 +1,23 @@
...
+    org: microsoft
+    tag: latest
```
Comment (unguiculus) [inline stable/msoms/values.yaml:10]: Can you specify a concrete version?

File: stable/msoms/templates/omsagent-daemonset.yaml [hunk 11]
```diff
@@ -0,0 +1,55 @@
+apiVersion: extensions/v1beta1
```
Comment (unguiculus) [inline stable/msoms/templates/omsagent-daemonset.yaml:1]: The file is incorrectly indented.

File: stable/msoms/templates/omsagent-secret.yaml [hunk 12]
```diff
@@ -0,0 +1,9 @@
+apiVersion: v1
+kind: Secret
+metadata:
+  name: {{ .Values.secretname }}
```
Comment (unguiculus) [inline stable/msoms/templates/omsagent-secret.yaml:4]: Don't make it configurable, just use `{{ template "fullname" . }}`

**Review by unguiculus — CHANGES_REQUESTED** [review]
* Please check best practices again
* Add missing `_helpers.tpl` and use it

File: stable/msoms/values.yaml (hunk 10 above)
Comment (weinong) [inline stable/msoms/values.yaml:10]: I'd suggest not to use `latest` tag especially with pull policy being Always.

**Comment by unguiculus** [comment]
Marking this as stale. Please update within one week.

**Comment by unguiculus** [comment]
Closing as stale. Feel free to reopen when/if you decide to work on it again.

**Pull request closed without merging**